gui/src/components/Nfts.tsx

```diff
@@ -0,0 +1,78 @@
+import { Card, CardContent, CardMedia, Grid, Typography } from "@mui/material";
+import { useState } from "react";
+
+import { useNfts } from "../store/nfts";
+import { Address, Nft } from "../types";
+import { Modal, NftDetailsView } from "./";
+import { AddressView } from "./AddressView";
+import { Panel } from "./Panel";
+
+interface ItemProps {
+  contract: Address;
+  imgSrc: string | undefined;
+  clickHandler: () => void;
+}
+
+export function Nfts() {
+  const [detailsViewOpen, setDetailsViewOpen] = useState(false);
+  const [currentNftDetails, setCurrentNftDetails] = useState<
+    Nft | undefined
+  >(undefined);
+
+  const nfts = useNfts((s) => s.nfts);
+
+  return (
+    <Panel>
+      <Grid container spacing={1}>
+        {nfts.map((nft) => {
+          const imgSrc = nft.metadata ? JSON.parse(nft.metadata).image : undefined;
+          return (
+            <Grid item key={`${nft.contract}-${ nft.token_id}`}>
+              <Item
+                contract={nft.contract}
+                clickHandler={() => {
+                  setCurrentNftDetails(nft);
+                }}
+                imgSrc={imgSrc} //nft?.imgSrc}
+              />
+            </Grid>
+          );
+        })}
+      </Grid>
+
+      <Modal
+        sx={{width: '80%'}}
+        open={detailsViewOpen}
+        onClose={() => {
+          setCurrentNftDetails(undefined);
+          setDetailsViewOpen(false);
+        }}
+      >
```

Comment:
I'm gonna let @gabrielpoca review the modal part since I'm not sure if:
1. this is the approach we want to go with?
2. the current logic plays well with the rest of the UI? (e.g.: what if we trigger the "command bar -> open settings" option while an NFT modal is open?

Reply:
For me, the Settings are a particular type of modal, like the Command Bar, that can be opened and closed from anywhere without messing up the app's navigation. I think this modal NFT should be a page (for instance, `/nfts/:id`) for now. Visually, it can be a modal on top of the NFT list, as long as the settings and command bar always show up on top, but making it a page makes it so we can link to it, refresh without losing state, and avoid future problems of having way too many places where modals can be open.